src/wrappers/withContainerDimensions/index.js

```diff
@@ -5,66 +5,71 @@ import { checkInsideGrid } from '../../utils';
  */
-const withContainerDimensions = (Component) => {
-  class WithContainerDimensions extends React.Component {
-    constructor(props) {
-      super(props);
-
-      this.state = {
-        referenceHeight: 0,
-        referenceWidth: 0,
-      };
-    }
+class WithContainerDimensions extends React.Component {
+  constructor(props) {
+    super(props);
 
-    componentWillMount() {
-      this.context.referenceSizeProvider.subscribe(this.handler);
-    }
+    this.state = {
+      referenceHeight: 0,
+      referenceWidth: 0,
+    };
+  }
 
-    componentWillUnmount() {
-      this.context.referenceSizeProvider.unsubscribe(this.handler);
-    }
+  componentWillMount() {
+    this.context.referenceSizeProvider.subscribe(this.handler);
+  }
 
-    handler = (width, height) => {
-      // Only update component on change.
-      if (height === this.state.referenceHeight
-        && width === this.state.referenceWidth) {
-        return;
-      }
-
-      this.setState({
-        referenceWidth: width,
-        referenceHeight: height,
-      });
-    }
+  componentWillUnmount() {
+    this.context.referenceSizeProvider.unsubscribe(this.handler);
+  }
 
-    render() {
-      return (
-        <Component
-          width={this.state.referenceWidth}
-          height={this.state.referenceHeight}
-          {...this.props}
-        />
-      );
+  handler = (width, height) => {
+    // Only update component on change.
+    if (height === this.state.referenceHeight
+      && width === this.state.referenceWidth) {
+      return;
     }
+
+    this.setState({
+      referenceWidth: width,
+      referenceHeight: height,
+    });
   }
 
-  const componentName = Component.displayName || Component.name || 'UnnamedComponent';
-  WithContainerDimensions.displayName = `withContainerDimensions(${componentName})`;
+  render() {
+    return (
+      this.props.children(this.state.referenceWidth, this.state.referenceHeight)
+    );
+  }
+}
 
-  WithContainerDimensions.contextTypes = {
-    referenceSizeProvider: checkInsideGrid(PropTypes.shape({
-      subscribe: PropTypes.func.isRequired,
-      unsubscribe: PropTypes.func.isRequired,
-    })),
-  };
+WithContainerDimensions.propTypes = {
+  children: PropTypes.func.isRequired,
+};
 
-  return WithContainerDimensions;
+WithContainerDimensions.contextTypes = {
+  referenceSizeProvider: checkInsideGrid(PropTypes.shape({
+    subscribe: PropTypes.func.isRequired,
+    unsubscribe: PropTypes.func.isRequired,
+  })),
 };
 
-export default withContainerDimensions;
+
+export const withContainerDimensions = (Component) => {
+  const withContainerDimensionsWrapped = props => (
+    <WithContainerDimensions>
+      {(width, height) => (<Component width={width} height={height} {...props} />)}
+    </WithContainerDimensions>);
+
+  const componentName = Component.displayName || Component.name || 'UnnamedComponent';
+  withContainerDimensionsWrapped.displayName = `withContainerDimensions(${componentName})`;
```

Comment:
Are this and the previous line even needed when creating wrapper HOC using FAAC? I would guess that by removing them it would just say anonymous function, but it would show that `WithSizeClass` is inside which wasn't the case when there was just wrapper.